packages/charts/react-charting/src/components/DeclarativeChart/PlotlySchemaAdapter.ts

```diff
@@ -2550,3 +2559,37 @@ const getBarProps = (
     xAxisOuterPadding: padding / 2,
   };
 };
+
+type GetAxisScaleTypePropsResult = Pick<ICartesianChartProps, 'xScaleType' | 'yScaleType' | 'secondaryYScaleType'>;
+
+const getAxisScaleTypeProps = (
+  data: Data[],
+  layout: Partial<Layout> | undefined,
+): Pick<ICartesianChartProps, 'xScaleType' | 'yScaleType' | 'secondaryYScaleType'> => {
+  const result: GetAxisScaleTypePropsResult = {};
+
+  const xAxisIds = new Set<number>();
+  const yAxisIds = new Set<number>();
+  data.forEach((series: Partial<PlotData>) => {
+    const axisIds = getAxisIds(series);
+    xAxisIds.add(axisIds.x);
+    yAxisIds.add(axisIds.y);
+  });
+
+  const axisKey = (axLetter: string, axId: number) => `${axLetter}axis${axId > 1 ? axId : ''}` as keyof Layout;
+
+  const sortedXAxisIds = Array.from(xAxisIds).sort();
+  if (sortedXAxisIds.length > 0 && layout?.[axisKey('x', sortedXAxisIds[0])]?.type === 'log') {
```

Comment:
>0 [](http://example.com/codeflow?start=72&length=1)

does this work if log axis is present in multiplot scenario which is like xaxis3, yaxis3 etc

Reply:
This is written assuming that the traces in a subplot use the same x-axis, as our cartesian charts do not support multiple x-axes.

Reply:
I think this function already handles the above scenario. We're grouping traces based on the `xaxis` property, and for each group/subplot, the adapter functions are invoked with the corresponding filtered data. This means all traces passed to an adapter function already share the same `xaxis`. This function then extracts the axis IDs from the traces and checks their type in the layout. That said, I could have just stored the xAxisId as a number instead of using a Set, which might be unnecessarily confusing.